### PR TITLE
traps can cast spells

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary

```SUMMARY: Features "traps can cast spells"```
#### Purpose of change

make traps able to cast spells!
#### Describe the solution

create a trapfunc called cast_spell and adds a fake_spell object to trap. creates a new trap that summons bears, called "bear trap". i didn't spawn it anywhere yet but i'm sure it can go in future areas.
#### Testing

used the map editor to spawn in a bear trap. walk over it and summon 3 bears.

### CODE REVIEW

**Comment by Funguss** [comment]
That is brilliant! Keep up the bear related humour!

Also, nice feature. :+1:

**Comment by Night-Pryanik** [comment]
I'm not familiar with summoning effect. Are summoned monsters friendly?

File: data/mods/Magiclysm/traps.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[
+  {
+    "type": "trap",
+    "id": "tr_bear",
+    "name": "bear trap",
+    "looks_like": "tr_beartrap",
```
Comment (BevapDin) [inline data/mods/Magiclysm/traps.json]: Traps don't support "looks_like". This causes the test to fail and will cause a debug message when the game starts.

Comment (KorGgenT) [reply data/mods/Magiclysm/traps.json]: ahh bummer. i'll remove it, though i definitely think it should gain this functionality

**Comment by KorGgenT** [comment]
> I'm not familiar with summoning effect. Are summoned monsters friendly?

these aren't, due to the HOSTILE_SUMMON flag, and they should not be by default because the trap passes basically a temporary brand new `npc` object, so it should basically not have a faction.

**Pull request merged**